nupm/install.nu

```diff
@@ -23,107 +23,161 @@ def throw-error [
                 log debug $"installing scripts for package ($package.name)"
-                install-scripts $path $package
+                $package.scripts | if $force {
+                    install-scripts --force $pkg_dir (script-dir --ensure)
+                } else {
+                    install-scripts $pkg_dir (script-dir --ensure)
+                }
             }
         },
         "script" => {
-            if "scripts" not-in $package {
-                let text = $"package is a script but does not have a `$.scripts` list"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            log debug $"installing scripts for package ($package.name)"
+
+            $package.scripts?
+            | default [ ($pkg_dir | path join $"($package.name).nu") ]
+            | if $force {
+                install-scripts --force $pkg_dir (script-dir --ensure)
+            } else {
+                install-scripts $pkg_dir (script-dir --ensure)
             }
-
-            install-scripts $path $package
         },
         "custom" => {
-            if not ($path | path join "build.nu" | path exists) {
-                let text = $"package uses a custom install but no `build.nu` has been found"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            let build_file = $pkg_dir | path join "build.nu"
+            if not ($build_file | path exists) {
+                let text = "package uses a custom install but no `build.nu` has"
+                        + " been found"
+                (throw-error
+                    "invalid_package_file"
+                    $text
+                    --span (metadata $pkg_dir | get span))
             }
 
-            nu ($path | path join 'build.nu')
+            let tmp_dir = tmp-dir build --ensure
+            cd $tmp_dir
+
+            nu $build_file ($pkg_dir | path join 'package.nuon')
+
+            cd -
+            rm -rf $tmp_dir
         },
         _ => {
-            let text = $"expected `$.type` to be one of [module, script, custom], got ($package.type)"
-            throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            let text = $"expected `$.type` to be one of [module, script,"
+                + " custom], got ($package.type)"
+            (throw-error
+                "invalid_package_file"
+                $text
+                --span (metadata $pkg_dir | get span))
         },
     }
 }
+
+# Install a nupm package
+export def main [
+    name    # Name, path, or link to the package
+    --path  # Install package from a directory with package.nuon given by 'name'
+    --force(-f)  # Overwrite already installed package
+] {
+    nupm-home-prompt
+
+    if not $path {
+        throw-error "`nupm install` currently requires a `--path` flag"
+    }
+
+    if $force {
+        install-path --force $name
+    } else {
+        install-path $name
+    }
```

Comment:
i think this can be simplified to
```nushell
install-path --force $force $name
```
if you change `install-path` to take `--force (-c): bool = false` :yum:

Reply:
Unfortunately, it doesn't. It would be super useful for flag passing, though.

Reply:
mm not sure this is correct anymore with https://github.com/nushell/nushell/pull/10456, right? :smirk: :yum:

Reply:
Done

Reply:
in https://github.com/nushell/nupm/pull/18/commits/f4a5e427394006da9302e6c64e88aa760a705974 :wink: